### PR TITLE
Ks renderer

### PR DESCRIPTION
**Pull request by escattone**
**DO NOT MERGE YET**

This is the PR that we'll merge once the Kumascript rendering work is in a minimum viable state. Opening this now (after re-basing on `master`) so the entire change will be visible. I'll merge `mdn:master` into this PR from time to time to keep it in sync with `master` until we merge it.

Don't let the `Files changed` count discourage, almost all of that is the addition of `kumascript` as a new yarn workspace.

### CODE REVIEW

**Comment by peterbe** [comment]
Just a drive by comment; can I kindly ask that you **merge** in upstream master instead. Then it’ll be easier for me to participate.

**Comment by escattone** [comment]
> Just a drive by comment; can I kindly ask that you **merge** in upstream master instead. Then it’ll be easier for me to participate.

@peterbe Will do, sorry about that! I was trying to do some re-organizing, and realized that I only made it worse.

File: .gitignore [hunk 1]
```diff
@@ -13,11 +13,13 @@
 
 # misc
 .DS_Store
+.env
 .env.local
 .env.development.local
 .env.test.local
 .env.production.local
 .env
+.vscode/
```
Comment (peterbe) [inline .gitignore:22]: As a general piece of advice, this is your personal preference and has nothing to do with stumptown-renderer. I would advance that you add to your `~/.gitignore_global` now that you're on the vs code band wagon. 
```
▶ cat ~/.gitignore_global | rg vscode
.vscode/
```

File: .gitignore [hunk 2]
```diff
@@ -13,11 +13,13 @@
 
 # misc
 .DS_Store
+.env
```
Comment (peterbe) [inline .gitignore:16]: Strange that you added this because line 21 (was line 20) already had it.

File: .gitignore (hunk 1 above)
Comment (escattone) [reply .gitignore:22]: Good point, will do!

File: .gitignore (hunk 2 above)
Comment (escattone) [reply .gitignore:16]: oops, i'll take that out!

File: .gitignore [hunk 3]
```diff
@@ -29,6 +31,9 @@ server/dist/
 ssr/node_modules/
 ssr/dist/
 watcher/node_modules/
+archivecontent/
 content/node_modules/
+content/coverage/
+content/files/
```
Comment (peterbe) [inline .gitignore:37]: I was hoping, that as soon as we get a somewhat decent ks renderer, that we can actually check in all the `content/files/**/index.html` and `content/files/**/index.yaml` files for good. 
But perhaps we can do that as a final hurray the day this PR merges into master.

File: content/package.json [hunk 4]
```diff
@@ -12,7 +12,10 @@
     "chokidar": "3.3.1",
     "dotenv": "8.2.0",
     "fast-csv": "4.1.3",
+    "got": "10.7.0",
+    "html-differ": "^1.4.0",
```
Comment (peterbe) [inline content/package.json:16]: Curious that this got added with a `^` but the `got` one above didn't. 
How did you add these two?

File: content/scripts/importer.js [hunk 5]
```diff
@@ -519,10 +520,7 @@ async function processDocument(
   await fs.promises.mkdir(folder, { recursive: true });
   const htmlFile = path.join(folder, "index.html");
 
-  // XXX As of right now, we don't have a KS shim that converts "raw Kuma HTML"
-  // to rendered HTML. So we'll cheat by copying the `rendered_html`.
-  // await fs.promises.writeFile(htmlFile, doc.html);
-  await fs.promises.writeFile(htmlFile, `${doc.rendered_html}`);
+  await fs.promises.writeFile(htmlFile, doc.html);
```
Comment (peterbe) [inline content/scripts/importer.js:523]: This line is sooo exciting!

File: .gitignore (hunk 3 above)
Comment (escattone) [reply .gitignore:34]: Sounds good!

File: kumascript/index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+const Templates = require("./src/templates.js");
+const AllPagesInfo = require("./src/info.js");
+
+class Renderer {
+  constructor() {
+    this.allPagesInfo = null;
+    this.templates = new Templates();
+  }
+
+  use(pageInfoByUri) {
```
Comment (peterbe) [inline kumascript/index.js:11]: Does there need to be an `update` method as well?

https://github.com/mdn/stumptown-renderer/issues/472 is about the fact that `this.allTitles` can and will mutate. It's closely linked to https://github.com/mdn/stumptown-renderer/issues/466 which is about consecutive builds when you restart it on the terminal. 

Yeah, we're going to need to mutate `this.allPagesInfo` too. We'll just have to make a mental note to remember to implement that.

File: kumascript/index.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+const renderMacros = require("./src/render.js");
```
Comment (peterbe) [inline kumascript/index.js:1]: I don't know what the exact best practices say but most other imports in this project use the import style *without* file extension. E.g https://github.com/mdn/stumptown-renderer/blob/master/content/index.js#L4
One immediate advantage I can think of is that we become unaffected if we change to TypeScript file extensions or something.

File: content/package.json (hunk 4 above)
Comment (escattone) [reply content/package.json:16]: Hm, I must have added `got` manually (shame on me!). I'll change that. Thanks!

Comment (peterbe) [reply content/package.json:16]: We actually have a think in CI that protects from that. So you would have gotten caught sooner or later :)

File: kumascript/index.js (hunk 6 above)
Comment (escattone) [reply kumascript/index.js:11]: The `use()` method can be called again whenever `this.allTitles` changes, and the render will use the updated values, so I think we're good here!

**Pull request closed without merging**